### PR TITLE
[L1] CSCTF fix issue with writing beyond array boundaries

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

Addressing issue https://github.com/cms-sw/cmssw/issues/37687. spInput returns a number between 1 and 15.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: EventFilter/CSCTFRawToDigi/plugins/CSCTFUnpacker.cc [hunk 1]
```diff
@@ -183,9 +183,9 @@ void CSCTFUnpacker::produce(edm::Event& e, const edm::EventSetup& c) {
               if (lct.empty())
                 continue;
 
-              status.link_status[lct[0].spInput()] |= (1 << lct[0].receiver_status_frame1()) |
-                                                      (1 << lct[0].receiver_status_frame2()) |
-                                                      ((lct[0].aligment_fifo() ? 1 : 0) << 4);
+              status.link_status[lct[0].spInput() - 1] |= (1 << lct[0].receiver_status_frame1()) |
```
Comment (perrotta) [inline EventFilter/CSCTFRawToDigi/plugins/CSCTFUnpacker.cc:186]: It is always dangerous to make differences of unsigned quantities.
On the other hand, CSCSP_MEblock::spInput_ is defined as being in the range [1-15], and the same difference is already used later on in this very same code in:
https://github.com/cms-sw/cmssw/blob/0f53d79f47b13a7ab5a1e709c5369e660e46ebba/EventFilter/CSCTFRawToDigi/plugins/CSCTFUnpacker.cc#L308
and
https://github.com/cms-sw/cmssw/blob/0f53d79f47b13a7ab5a1e709c5369e660e46ebba/EventFilter/CSCTFRawToDigi/plugins/CSCTFUnpacker.cc#L309

I assume therefore which is also safe to have the same difference computed here

**Comment by perrotta** [comment]
+1

**Pull request merged**